agent/hcp/client/client.go

```diff
@@ -20,15 +22,34 @@ import (
 	"github.com/hashicorp/consul/version"
 )
 
+// metricsGatewayPath is the default path for metrics export request on the Telemetry Gateway.
+const metricsGatewayPath = "/v1/metrics"
+
 // Client interface exposes HCP operations that can be invoked by Consul
 //
 //go:generate mockery --name Client --with-expecter --inpackage
 type Client interface {
 	FetchBootstrap(ctx context.Context) (*BootstrapConfig, error)
+	FetchTelemetryConfig(ctx context.Context) (*TelemetryConfig, error)
```

Comment:
Note for the future. Try not to grow interfaces as much as possible. The larger the interface the weaker the abstraction. It also stops you from making composable implementations and forces you to build a mock for FetchBootstrap, when you might only want to test FetcyTelemetryConfig.

Reply:
Notes! I was trying to use the same pattern, but agreed.